File: src/main/java/com/dorjear/training/fundamental/hackerrank/Duplicates.java
package com.dorjear.training.fundamental.hackerrank;

import java.util.Map;
import java.util.stream.Collectors;

public class Duplicates {
    private static long duplicateCount(String input) {
        return input
                .toLowerCase()
                .chars()
                .mapToObj(c -> (char) c)
                .collect(
                    Collectors.toMap(c -> c, c -> 1, Integer::sum)
                )
                .entrySet().stream().filter(entry -> entry.getValue()>=2).count();
    }

    public static void main(String[] args) {
        System.out.println(duplicateCount("abcde"));
        System.out.println(duplicateCount("aabbcde"));
        System.out.println(duplicateCount("aabBcde"));
        System.out.println(duplicateCount("indivisibility"));
        System.out.println(duplicateCount("Indivisibilities"));
    }
}

File: src/main/java/com/dorjear/training/fundamental/kotlin/Duplicates.kt
package com.dorjear.training.fundamental.kotlin

import java.util.*
import java.util.function.BinaryOperator
import java.util.function.Function
import java.util.stream.Collectors

object Duplicates {
    private fun duplicateCount(input: String): Int {
        return input
                .lowercase()
                .chars()
                .mapToObj { c: Int -> c.toChar() }
                .collect(
                        Collectors.toMap({ c: Char? -> c }, { 1 }, { a: Int, b: Int -> Integer.sum(a, b) })
                )
                .values.count { it > 1 }
    }

    @JvmStatic
    fun main(args: Array<String>) {
        println(duplicateCount("abcde"))
        println(duplicateCount("aabbcde"))
        println(duplicateCount("aabBcde"))
        println(duplicateCount("indivisibility"))
        println(duplicateCount("Indivisibilities"))
    }
}